web/client/components/data/featuregrid/renderers/CellRenderer.jsx

```diff
@@ -24,7 +24,7 @@ class CellRenderer extends React.Component {
     }
     render() {
         const isProperty = this.context.isProperty(this.props.column.key);
-        const isModified = (this.props.rowData._new && isProperty) || this.context.isModified(this.props.rowData.id, this.props.column.key);
+        const isModified = (this.props.rowData._new && isProperty);
```

Comment:
@mulondo I think we can't solve the issue by removing the [check for modified values](https://github.com/geosolutions-it/MapStore2/blob/e97942967045fef363cd8b8a36a49a36b950f8a5/web/client/components/data/featuregrid/FeatureGrid.jsx#L78-L81) as you are proposing here because this will remove the highlights also for the one already working.

@tdipisa I did an investigation on the reason of the highlighting of changed fields even if there are no visual changes from a UI. The reason seems related to String and Boolean property with `null` value (new created feature has all properties set to null).  When we are changing the value from the UI for string and boolean we are transforming a `null` value to `""` (an empty string) so this is detected has change and highlighted, even if we are rendering `null` and `""` in the same way inside the grid cell.

I think we can solve the problem of the boolean value by adding a fallback to null instead of empty string while we are [forcing the selection](https://github.com/geosolutions-it/MapStore2/blob/e97942967045fef363cd8b8a36a49a36b950f8a5/web/client/utils/FeatureGridEditorUtils.js#L12) but for the string is a bit different. I wondering if from the UI should be clear when a field is `null` because an empty string it could be a valid input for this type of property, if so I think the issue need to be reviewed and blocked for the moment

Reply:
I would like to concentrate the effort on the _boolean_ case for now. Then, through a dedicated issue, we could think on a way to graphically distinguish a _Null_ String from an _Empty_ string and maybe apply that graphic enhancement to all Null fields independently on their type: I think we should dedicate some of our time to evaluate the "how to" better.

Reply:
> @mulondo I think we can't solve the issue by removing the [check for modified values](https://github.com/geosolutions-it/MapStore2/blob/e97942967045fef363cd8b8a36a49a36b950f8a5/web/client/components/data/featuregrid/FeatureGrid.jsx#L78-L81) as you are proposing here because this will remove the highlights also for the one already working.
> 
> @tdipisa I did an investigation on the reason of the highlighting of changed fields even if there are no visual changes from a UI. The reason seems related to String and Boolean property with `null` value (new created feature has all properties set to null). When we are changing the value from the UI for string and boolean we are transforming a `null` value to `""` (an empty string) so this is detected has change and highlighted, even if we are rendering `null` and `""` in the same way inside the grid cell.
> 
> I think we can solve the problem of the boolean value by adding a fallback to null instead of empty string while we are [forcing the selection](https://github.com/geosolutions-it/MapStore2/blob/e97942967045fef363cd8b8a36a49a36b950f8a5/web/client/utils/FeatureGridEditorUtils.js#L12) but for the string is a bit different. I wondering if from the UI should be clear when a field is `null` because an empty string it could be a valid input for this type of property, if so I think the issue need to be reviewed and blocked for the moment

@allyoucanmap I am done with that!